src/components/totals/Totals.tsx

```diff
@@ -0,0 +1,199 @@
+  const [ChartDataIsLoading, setChartDataIsLoading] = useState(true);
+  const [TotalsData, setTotals] = useState<TotalDataDTO[]>([]);
+  const [Currencies, setCurrencies] = useState([]);
+  const [ChartData, setChartData] = useState<ChartDataDTO[]>([]);
+  const [Currency, setCurrency] = useState(1);
+  const [ChartScale, setChartScale] = useState<'day' | 'week' | 'month'>('day');
+  //default date is last 7 days
+  const [body, setbody] = useState<RequestBodyDTO>({
+    start_date: new Date(
+      new Date().getFullYear(),
+      new Date().getMonth(),
+      new Date().getDate() - 7
+    )
+      .toISOString()
+      .split('T')[0],
+    end_date: new Date().toISOString().split('T')[0],
+  });
+
+  const handleDateChange = (DateSet: string) => {
+    const data = DateSet.split(',');
+    const s_date = new Date(DateSet.split(',')[0]);
+    const e_date = new Date(DateSet.split(',')[1]);
+
+    if (s_date > e_date) {
+      setbody({
+        start_date: data[1],
+        end_date: data[0],
+      });
+    } else {
+      setbody({ start_date: data[0], end_date: data[1] });
+    }
+  };
+  const getTotals = async () => {
+    await request(
+      'POST',
+      endpoint('/api/v1/charts/totals'),
+      body,
+      defaultHeaders
+    ).then((response: AxiosResponse) => {
+      setTotals(response.data);
+      setCurrencies(response.data.currencies);
+
+      setTotalsIsLoading(false);
+    });
+  };
+
+  const getChartData = async () => {
+    await request(
+      'POST',
+      endpoint('/api/v1/charts/chart_summary'),
+      body,
+      defaultHeaders
+    ).then((response: AxiosResponse) => {
+      setChartData(response.data);
+      setChartDataIsLoading(false);
+    });
+  };
+
+  useEffect(() => {
+    getTotals();
+    getChartData();
+  }, [body]);
+  return (
+    <div className="max-w-screen-2xl mx-auto">
+      {TotalsIsLoading && <Spinner />}
+
+      {!TotalsIsLoading && (
+        <div className="flex flex-col sm:grid grid-cols-3 gap-5 my-5 ">
+          <div className="flex flex-wrap justify-center col-start-2">
+            {Currencies &&
+              Object.entries(Currencies).map(([key, value]) => {
+                return (
+                  <Button
+                    key={`currency-btn-${key}`}
+                    value={value}
+                    type="secondary"
+                    className={'mx-0.5'}
+                    onClick={() => {
+                      setCurrency(Number(key));
+                    }}
+                  >
+                    {String(value)}
+                  </Button>
+                );
+              })}
+          </div>
+          <div className="flex justify-center sm:justify-start">
+            <Button
+              key={`day-btn`}
+              className={'mx-0.5'}
+              type="secondary"
+              onClick={() => {
+                setChartScale('day');
+              }}
+            >
+              Day
```

Comment:
Please make use of translations. App itself is multi-language, so this ain't gonna cut it.